torchdynamo/_guards.cpp

```diff
@@ -59,6 +60,45 @@ class TensorCheck {
     return true;
   }
 
+  std::string check_verbose(const LocalState &state, const at::Tensor &v) {
```

Comment:
yes.. this is ugly.

i wasn't sure if it is worth expanding all the ||'d conditions into individual ones to get precise messages (yes?)

or if it is worth keeping this function totally separate from the main check function to avoid introducing more branches into it (also yes?)